### DISCUSSION TITLE
Download Self Hoster Runner binary from GitHub Server

### DISCUSSION THREAD

**Discussion by adammontlake** [community/community#52933] [Enterprise]
### Select Topic Area

Question

### Body

Hey community

Need some advice with setting up a self hosted runner in an isolate environment with GitHub Enterprise Server.
I can get a new runner registration token from the api: "api.git_url/repos/OWNER/REPO/actions/runners/registration-token"
and for the Authorization Bearer I am using a PAT.

Now I need to download the Runner binary: git_url/_services/pipelines/_apis/distributedtask/packagedownload/agent/linux-x64/VERSION
However, when using the PAT for authorization it comes back with an error: "The user ... is not authorized to access this resource."
When looking at the "settings/actions/runners/new" api, it returns in the response body a "data-download-token" which is authorized to download.

The question:
Besides parsing the response body, is there a better way to get this data-download-token from the api? Or a different way of downloading the runner binary with my PAT?

**Comment by charlysotelo** [comment]
Did you ever find a solution for this?

**Reply by adammontlake** [reply]
Yup, here is the relevant part of the code. If you download from the context of the repo there is no longer a permissions issue.
I uploaded a gist [here](https://gist.github.com/adammontlake/7bee567951a8d25e5a01977303fb9c8a).

**Reply by charlysotelo** [reply]
Yo you are da homie!
